pwiz_tools/Skyline/Test/SrmSettingsTest.cs

```diff
@@ -1185,109 +1186,112 @@ public void SerializeProteinAssociationSettingsTest()
-            Assert.AreEqual(100, pred.FilterWindowWidthCalculator.ResolvingPower);
-            CheckIonMobilitySettingsBackwardCompatibility(predictor3.Replace("20", "-1"),Resources.DriftTimeWindowWidthCalculator_Validate_Peak_width_must_be_non_negative_);
-            CheckIonMobilitySettingsBackwardCompatibility(predictor3.Replace("500", "-1"), Resources.DriftTimeWindowWidthCalculator_Validate_Peak_width_must_be_non_negative_);
-            Assert.AreEqual(widthAtDt0 + (widthAtDtMax-widthAtDt0)*driftTime/driftTimeMax, pred.FilterWindowWidthCalculator.WidthAt(driftTime, driftTimeMax));
-
-            // Test ability to roundtrip to older doc formats
-            var xml = SETTINGS_V19.Replace("</peptide_prediction>", predictor3 + "\n</peptide_prediction>");
-            var settings = AssertEx.Deserialize<SrmSettings>(xml);
-            var save = AuditLogList.IgnoreTestChecks;
-            AuditLogList.IgnoreTestChecks = true;
-            var tmpFile19 = "V19_1.sky";
-            var tmpFileCurrent = "V20_13.sky";
-            var oldDoc = new SrmDocument(settings.ChangeDataSettings(settings.DataSettings.ChangeAuditLogging(false)));
-            var testPath = TestContext.GetTestResultsPath();
-            if (!File.Exists(testPath))
-                Directory.CreateDirectory(testPath);
-            AssertEx.Serializable(oldDoc, testPath, SkylineVersion.V19_1); // Round trip with IMS info in peptide settings
-            AssertEx.Serializable(oldDoc, testPath, SkylineVersion.CURRENT); // Round trip with IMS info in transition settings
-            oldDoc.SerializeToFile(tmpFile19, tmpFile19, SkylineVersion.V19_1, null);
-            oldDoc.SerializeToFile(tmpFileCurrent, tmpFileCurrent, SkylineVersion.CURRENT, null);
-            var oldDocXML = File.ReadAllText(tmpFile19);
-            var currentDocXML = File.ReadAllText(tmpFileCurrent);
-            Assert.IsTrue(oldDocXML.Contains(DriftTimePredictor.EL.predict_drift_time.ToString()));
-            Assert.IsTrue(!currentDocXML.Contains(DriftTimePredictor.EL.predict_drift_time.ToString()));
-            var newDoc = AssertEx.Deserialize<SrmDocument>(oldDocXML);
-            var currentDoc = AssertEx.Deserialize<SrmDocument>(currentDocXML);
-            var diff = SrmDocument.EqualsVerbose(oldDoc, newDoc);
-            if (diff != null)
-                Assert.Fail(diff);
-            Assert.AreEqual(newDoc.Settings.TransitionSettings.IonMobilityFiltering.IonMobilityLibrary.Name, "test");
-            Assert.AreEqual(currentDoc.Settings.TransitionSettings.IonMobilityFiltering.IonMobilityLibrary.Name, "test");
-            AuditLogList.IgnoreTestChecks = save;
-
+            // Make sure any generated .imsdb files are not in exe's directory - that interferes with parallel tests
+            TestContext.EnsureTestResultsDir();
+            using (var d = new CurrentDirectorySetter(TestContext.GetTestResultsPath()))

```

Comment:
You can avoid all this whitespace churn with a using statement:
```
using var d = new CurrentDirectorySetter(TestContext.GetTestResultsPath());
```

Reply:
Thanks. I'm obviously unclear on the scoping rules around IDisposable.